### PR TITLE
StreamResultEncodingOverride

### PR DESCRIPTION
**Pull request by nhennebe67**

### CODE REVIEW

**Comment by nhennebe67** [comment]
ability to override XmlReader xml parsing Encoding

File: src/Splunk.Client/Splunk/Client/Response.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ namespace Splunk.Client
     using System.Net.Http;
     using System.Threading.Tasks;
     using System.Xml;
+	using System.Text;
```
Comment (shakeelmohamed) [inline src/Splunk.Client/Splunk/Client/Response.cs]: Please replace all tab character indentations with 4 spaces

File: src/Splunk.Client/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -25,9 +25,9 @@
 [assembly: AssemblyTitle("Splunk.Client")]
```
Comment (shakeelmohamed) [inline src/Splunk.Client/Properties/AssemblyInfo.cs]: Revert all changes to this file

File: splunk-sdk-csharp-pcl.sln [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (shakeelmohamed) [inline splunk-sdk-csharp-pcl.sln]: Revert all changes to this file

File: src/Splunk.Client/Splunk/Client/Response.cs [hunk 4]
```diff
@@ -73,13 +82,21 @@ public Stream Stream
...
+					}
+					else
+					{
+						this.reader = XmlReader.Create(this.Stream, XmlReaderSettings);
+					}
+
+				}
 
-                return this.reader;
+				return this.reader;
```
Comment (shakeelmohamed) [inline src/Splunk.Client/Splunk/Client/Response.cs]: Tabs

File: src/Splunk.Client/Splunk/Client/Response.cs [hunk 5]
```diff
@@ -63,6 +64,14 @@ public HttpResponseMessage Message
         public Stream Stream
         { get; private set; }
 
+		/// <summary>
+		/// Override the encoding used to parse the Reponse to prevent parsing exception.
+		/// This is useful if the response contains illegal characters that do not comply with the Xml Prolog.
+		/// For example the response contains unicode, whereas xml Prolog is <?xml version='1.0' encoding='UTF-8'?>
+		/// Leave null to use xml Prolog encoding.
+		/// </summary>
+		public static Encoding StreamResultEncodingOverride { get; set; }
```
Comment (shakeelmohamed) [inline src/Splunk.Client/Splunk/Client/Response.cs]: I think this property name is verbose, what about `StreamReaderEncoding`?

**Comment by shakeelmohamed** [comment]
Thanks for the PR @nhennebe67! I've left a few comments.

Please also:

0. Fill out the [individual contributor agreement](http://dev.splunk.com/goto/individualcontributions)
0. Submit a new PR against the `develop` branch

**Comment by nhennebe67** [comment]
Done created a new PR against develop, "Stream reader encoding #48".
Submitted individual contributor agreement

**Pull request closed without merging**